src/storage/resourceManager.ts

```diff
@@ -309,69 +311,87 @@ export class ResourceManager {
+  async getSchemaRegistries<T extends SchemaRegistry>(connectionId: ConnectionId): Promise<T[]> {
+    const storageKey = this.getSchemaRegistryKey(connectionId);
+
+    let registriesByConnectionString: string | undefined;
+
     // Get the JSON-stringified map from storage
-    const registriesByEnvString: string | undefined = this.workspaceState.get(
-      WorkspaceStorageKeys.CCLOUD_SCHEMA_REGISTRIES,
-    );
-    const registriesByEnv: Map<string, object> = registriesByEnvString
-      ? stringToMap(registriesByEnvString)
-      : new Map<string, object>();
-    // cast any values back to CCloudSchemaRegistry instances
-    return new Map(
-      Array.from(registriesByEnv).map(([envId, registry]) => [
-        envId,
-        CCloudSchemaRegistry.create(registry as CCloudSchemaRegistry),
-      ]),
-    );
-  }
+    registriesByConnectionString = await this.workspaceState.get(storageKey);
 
-  /**
-   * Get a specific Schema Registry from extension state.
-   * @param environmentId The ID of the {@link CCloudEnvironment} from which to get the Schema Registry
-   * @returns The associated {@link CCloudSchemaRegistry}, or `null` (if the environment is not found or has no Schema Registry)
-   */
-  async getCCloudSchemaRegistry(environmentId: string): Promise<CCloudSchemaRegistry | null> {
-    const schemaRegistries: CCloudSchemaRegistryByEnv = await this.getCCloudSchemaRegistries();
-    const schemaRegistryForEnv: CCloudSchemaRegistry | null =
-      schemaRegistries.get(environmentId) ?? null;
-    if (!schemaRegistryForEnv) {
-      logger.warn(`No Schema Registry found for environment ${environmentId}`);
-    }
-    return schemaRegistryForEnv;
-  }
+    const registriesByConnection: Map<string, object[]> = registriesByConnectionString
+      ? stringToMap(registriesByConnectionString)
+      : new Map<string, object[]>();
 
-  /** Get a specific Schema Registry by its id */
-  async getCCloudSchemaRegistryById(id: string): Promise<CCloudSchemaRegistry | null> {
-    const clusters = await this.getCCloudSchemaRegistries();
-    for (const cluster of clusters.values()) {
-      if (cluster.id === id) {
-        return cluster;
-      }
+    // Will either be undefined or an array of plain objects since just deserialized from storage.
+    const vanillaJSONRegistries: object[] | undefined = registriesByConnection.get(connectionId);
+
+    if (!vanillaJSONRegistries) {
+      return [];
     }
-    return null;
+
+    logger.debug(
+      `Found ${vanillaJSONRegistries.length} schema registries for connection ${connectionId}`,
+    );
+
+    const schemaRegistryClass = SchemaRegistry.getSchemaRegistryClass(connectionId);
+    // Promote each object member to be the proper instance of SchemaRegistry sub-type, return.
+    return vanillaJSONRegistries.map((registry) => schemaRegistryClass.create(registry as T) as T);
```

Comment:
Feels like we could combine these and consolidate the static method on SchemaRegistry to take a connectionId and the schema registry data/object to create the right instance 🤔